Missing esm build file

Using 3.0.3 I see an error with import random from "random" in a Create React App project.
It says:

./node_modules/random/dist/esm/index.js
Module not found: Can't resolve './random' in './node_modules/random/dist/esm'

I've double checked: there is an index.js there, it tries to require('./random').default but fails because there isn't any random.js file in there :(

I believe it might be connected to #19

--- a/.gitignore
+++ b/.gitignore
@@ -6,6 +6,7 @@ node_modules
 # builds
 build
 dist
+tsconfig.tsbuildinfo
 
 # misc
 .DS_Store

--- a/package.json
+++ b/package.json
@@ -3,7 +3,7 @@
   "version": "3.0.3",
   "description": "Seedable random number generator supporting many common distributions.",
   "main": "dist/cjs/index.js",
-  "module": "dist/esm/index.js",
+  "module": "dist/esm/index.esm.js",
   "types": "dist/cjs/index.d.ts",
   "repository": "transitive-bullshit/random",
   "author": "Travis Fischer <travis@transitivebullsh.it>",

--- a/src/index.esm.ts
+++ b/src/index.esm.ts
@@ -1,0 +1,3 @@
+import random from './random'
+export * from './random'
+export default random

--- a/tsconfig.cjs.json
+++ b/tsconfig.cjs.json
@@ -1,5 +1,6 @@
 {
   "extends": "./tsconfig.json",
+  "exclude": ["**/index.esm.ts"],
   "compilerOptions": {
     "noEmit": false,
     "module": "commonjs",

--- a/tsconfig.esm.json
+++ b/tsconfig.esm.json
@@ -2,7 +2,7 @@
   "extends": "./tsconfig.json",
   "compilerOptions": {
     "noEmit": false,
-    "module": "commonjs",
-    "outDir": "dist/cjs"
-  }
+    "module": "es6",
+    "outDir": "dist/esm"
+  },
 }

--- a/tsconfig.esm.json
+++ b/tsconfig.esm.json
@@ -1,5 +1,6 @@
 {
   "extends": "./tsconfig.json",
+  "exclude": ["**/index.ts"],
   "compilerOptions": {
     "noEmit": false,
     "module": "es6",

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -2,15 +2,14 @@
   "compilerOptions": {
     "composite": true,
     "target": "es6",
-    "module": "commonjs",
     "moduleResolution": "node",
     "allowJs": true,
     "declaration": true,
     "declarationMap": true,
     "sourceMap": true,
-    "outDir": "dist/esm",
     "rootDir": "src",
     "strict": true,
+    "noEmit": true,
     "noImplicitAny": true,
     "strictNullChecks": true,
     "strictFunctionTypes": true,
@@ -26,11 +25,14 @@
     "esModuleInterop": true,
     "forceConsistentCasingInFileNames": true
   },
-  "include": ["src/**/*"],
+  "include": ["src"],
   "exclude": ["node_modules", "test"],
   "references": [
     {
       "path": "./tsconfig.cjs.json"
+    },
+    {
+      "path": "./tsconfig.esm.json"
     }
   ]
 }

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -25,7 +25,7 @@
     "esModuleInterop": true,
     "forceConsistentCasingInFileNames": true
   },
-  "include": ["src"],
+  "include": ["src/**/*"],
   "exclude": ["node_modules", "test"],
   "references": [
     {

